validation/src/lib.rs

```diff
@@ -354,38 +371,41 @@ impl<C, N, P> ParachainValidation<C, N, P> where
 				validation_para,
 				relay_parent,
 				collators,
-				client,
+				client.clone(),
 				max_block_data_size,
 			);
 
 			collation_work.then(move |result| match result {
-				Ok((collation, outgoing_targeted)) => {
-					let outgoing_queues = crate::outgoing_queues(&outgoing_targeted)
-						.map(|(_target, root, data)| (root, data))
-						.collect();
-
-					let res = availability_store.make_available(Data {
-						relay_parent,
-						parachain_id: collation.receipt.parachain_index,
-						candidate_hash: collation.receipt.hash(),
-						block_data: collation.pov.block_data.clone(),
-						outgoing_queues: Some(outgoing_queues),
-					});
-
-					match res {
-						Ok(()) => {
-							// TODO: https://github.com/paritytech/polkadot/issues/51
-							// Erasure-code and provide merkle branches.
-							router.local_collation(collation, outgoing_targeted);
+				Ok((collation, outgoing_targeted, fees_charged)) => {
+					match produce_receipt_and_chunks(
+						authorities_num,
+						&relay_parent,
+						&collation.pov,
+						&outgoing_targeted,
+						fees_charged,
+						&collation.info,
+					) {
+						Ok((receipt, chunks)) => {
+							let res = availability_store.add_erasure_chunks(chunks.clone());
```

Comment:
so that the chunk data doesn't have a lot of redundancy, we should just have the availability store function accept the candidate receipt as an argument

Reply:
when adding a single chunk, the index should be passed alongside it